### PR TITLE
Windows home/end shortcuts

### LINKED ISSUE #77020: TextField: Missing keyboard shortcuts for Windows
I follow recomendation in https://github.com/flutter/flutter/issues/64721#issuecomment-712919337 (**"If you find specific issues in a current version..."** )

For windows, the following shortcuts are missing: Ctrl+End, Ctrl+Home, Ctrl+Shift+End, Ctrl+Shift+Home, Shift+End, Shift+Home

### PR DESCRIPTION
**Pull request by justinmc**
This PR fixes some keyboard shortcuts on Windows that were missing/incorrect.  From the issue, these were my observations before starting this PR:

| Shortcut | Behavior (Windows) | Behavior (Flutter) |
| -- | -- | -- |
| Ctrl+Home/End | move to start/end of all text | ❌ nothing |
| Ctrl+Shift+Home/End | extend selection to start/end of all text | ❌ nothing |
| Shift+Home/End | extend selection to start/end of line | ❌ works, but pauses at inversion, and jumps lines |

Fixes https://github.com/flutter/flutter/issues/77020
I've opened a separate issue for the incorrect movement between lines: https://github.com/flutter/flutter/issues/90993

### CODE REVIEW

**Comment by justinmc** [comment]
@LongCatIsLooong FYI this is going to conflict with your refactor https://github.com/flutter/flutter/pull/90684.  I'm still working on this a bit so maybe I should just wait until your PR is merged and then make these changes in the new pattern.

**Comment by goderbauer** [comment]
From triage: Looks like #90684 is submitted now 🥳

**Comment by justinmc** [comment]
Will update this, thanks!

File: packages/flutter/lib/src/widgets/editable_text.dart
Comment (justinmc) [inline packages/flutter/lib/src/widgets/editable_text.dart:3652]: @LongCatIsLooong How would you prefer that we implement this "collapseAtReversal" feature within your new Actions setup?  Feel free to butcher this if it's totally off, I'm still getting the hang of your new refactor so I've just put down a quick attempt.

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:255]: I tried on windows 10, it doesn't seem to collapse at reversal? E.g., the initial selection is `aaaaaa|bbbbb`, I pressed <kbd>shift</kbd>-<kbd>End</kbd> it became `aaaaaa<bbbbb>` and then I pressed <kbd>shift</kbd>-<kbd>Home</kbd> it became `<aaaaaa>bbbbb`. This was tested in the explorer's address bar.

Comment (justinmc) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart]: Thanks good catch.  My table in the description of this PR is correct, but I got this code backwards when I ported it.  I've fixed it.

I also double checked Linux (gedit).  It has the same behavior as Windows.

**Comment by justinmc** [comment]
@LongCatIsLooong In need of a re-review when you have a chance.

File: packages/flutter/lib/src/widgets/editable_text.dart
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/editable_text.dart:3652]: Yeah this makes sense to me!

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/test/widgets/default_text_editing_actions_test.dart
Comment (LongCatIsLooong) [inline packages/flutter/test/widgets/default_text_editing_actions_test.dart:76]: I thought I removed this file, since `TextEditingActionTarget` is gone?

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart
Comment (justinmc) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart]: I've reworked this to just use `..._commonShortcuts` here.  I had to duplicate 4 entries but I think that's fine.

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart
Comment (justinmc) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:409]: Yeah that might help us avoid a lot of this code.  @gspencergoog Do you have any opinion?

File: packages/flutter/test/widgets/default_text_editing_actions_test.dart
Comment (justinmc) [reply packages/flutter/test/widgets/default_text_editing_actions_test.dart:76]: So did I but I also noticed that text_editing_action_target.dart was still in the repo unused!  I removed it in this PR.  Maybe it accidentally came back in a merge or something and we missed it?

I went ahead and deleted this test file too now.

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:409]: We could make a different constructor for Shortcuts that takes a function instead of a Map. I'd want to keep the one that takes a Map, though. One additional benefit of the function over the map would be that it might be faster to construct: long maps can get slow to construct on each build.

Comment (justinmc) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:409]: That would work, so all of this _webShortcuts map would just be a check for kIsWeb and if so return DoNothingAndStopPropagationTextIntent.

**Pull request merged**
